### PR TITLE
fix: autocomplete - first item active/tabbing

### PR DESCRIPTION
**Pull request by mishahoo-zz**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [ ] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [ ] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

My code does...

### CODE REVIEW

File: src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx [hunk 1]
```diff
@@ -176,6 +178,30 @@ export const ResultsContainer = forwardRef(
...
+                    return;
+                });
+            }
+            // If highlightFirst is true and term does not exist in results
+            // update termIndex to the first result
+            if (highlightFirst && !termIndex) {
+                // If first item is a title, skip to next item
+                termIndex = results[0].value.listItemType === 'title' ? 2 : 1;
+            }
+            // Set term index if it exists alond with initial term
```
Comment (mishahoo-zz) [inline src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx]: just noticed this misspell `alond` good one Misha

File: src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx [hunk 2]
```diff
@@ -176,6 +178,30 @@ export const ResultsContainer = forwardRef(
...
 
+        useEffect(() => {
+            // Check if current term exists in results and capture the index
+            let termIndex;
+            if (term) {
+                results.some((result: any, index: number) => {
+                    if (result.label === term) {
+                        return (termIndex = relativeIndex(index));
+                    }
+                    return;
```
Comment (zero0Halo) [inline src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx]: Is this `return` statement doing anything?

File: src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx [hunk 3]
```diff
@@ -227,24 +253,40 @@ export const ResultsContainer = forwardRef(
             }
         };
 
+        const isInRange = currentIndex >= 0 && currentIndex <= results.length;
+        // If currentIndex is 0, it's referring to the input field
+        const adjustedForInputIndex = currentIndex === 0 ? 0 : currentIndex - 1;
```
Comment (zero0Halo) [inline src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx]: I believe you can simplify this with `currentIndex ? currentIndex -1 : 0`. Could probably make it even more terse, but then it'd be hard to read.

File: src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx [hunk 4]
```diff
@@ -227,24 +253,40 @@ export const ResultsContainer = forwardRef(
...
+        // Check for a title item index and skip it
+        const checkForTitleIndex = (index: number) =>
+            results[index].value.listItemType === 'title' ? index + 1 : index;
```
Comment (zero0Halo) [inline src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx]: I think there could be a danger here if the last item in a list is of type `title`. `index + 1` wouldn't exist.

File: src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx [hunk 5]
```diff
@@ -176,6 +178,30 @@ export const ResultsContainer = forwardRef(
             setInitialTerm
         );
 
+        useEffect(() => {
+            // Check if current term exists in results and capture the index
+            let termIndex;
+            if (term) {
+                results.some((result: any, index: number) => {
```
Comment (zero0Halo) [inline src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx]: I think that [`findIndex`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) might be more appropriate, though I do like the readability here.

File: src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx (hunk 2 above)
Comment (mishahoo-zz) [reply src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx]: oh I was getting a linting error without it

File: src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx (hunk 5 above)
Comment (mishahoo-zz) [reply src/AutoComplete/ResultsContainer/ResultsContainer.component.tsx]: oh yes. this is much better.

**Review by zero0Halo — APPROVED** [review]
LGTM

**Pull request merged**